napari/experimental/_tests/test_progressive_loading.py

```diff
@@ -0,0 +1,330 @@
+
+
+@pytest.mark.xfail(
+    reason='shouldnt fail, the width in set_interval is calculated to be 1024, therefore the first and second hyperslice are equal'
+)
+def test_virtualdata_hyperslice_fail(mandelbrot_arrays, max_level):
+    scale = max_level - 1
+    vdata = _progressive_loading.VirtualData(
+        mandelbrot_arrays[scale], scale=scale
+    )
+    coords = (slice(0, 1024, None), slice(0, 1024, None))
+    vdata.set_interval(coords)
+    first_hyperslice = vdata.hyperslice
+    coords = (slice(512, 1024, None), slice(512, 1024, None))
+    # the width in set_interval is calculated to be 1024, therefore the first and second hyperslice are equal, causing this test to fail
+    vdata.set_interval(coords)
+    second_hyperslice = vdata.hyperslice
+    assert_raises(
+        AssertionError, assert_array_equal, first_hyperslice, second_hyperslice
+    )
+
+
+def test_virtualdata_hyperslice(mandelbrot_arrays, max_level):
+    scale = max_level - 1
+    vdata = _progressive_loading.VirtualData(
+        mandelbrot_arrays[scale], scale=scale
+    )
+    coords = (slice(0, 1024, None), slice(0, 1024, None))
+    vdata.set_interval(coords)
+    first_hyperslice = vdata.hyperslice
+    coords = (slice(513, 1024, None), slice(513, 1024, None))
+    # the width in set_interval is calculated to be 1024, therefore the first and second hyperslice are equal, causing this test to fail
+    vdata.set_interval(coords)
+    second_hyperslice = vdata.hyperslice
+    assert_raises(
+        AssertionError, assert_array_equal, first_hyperslice, second_hyperslice
+    )
+
+
+def test_multiscalevirtualdata_init(mandelbrot_arrays):
+    mvdata = _progressive_loading.MultiScaleVirtualData(mandelbrot_arrays)
+    assert isinstance(mvdata, _progressive_loading.MultiScaleVirtualData)
+
+
+def test_multiscalevirtualdata_set_interval(mandelbrot_arrays):
+    mvdata = _progressive_loading.MultiScaleVirtualData(mandelbrot_arrays)
+    coords = (slice(513, 1024, None), slice(513, 1024, None))
+    min_coord = [st.start for st in coords]
+    max_coord = [st.stop for st in coords]
+    mvdata.set_interval(min_coord=min_coord, max_coord=max_coord)
+
+    # mvdata slices are based on chunks so we make sure that the requested
+    # coordinates fall within the chunked slice
+    assert mvdata._data[0]._min_coord <= min_coord
+    assert mvdata._data[0]._max_coord <= max_coord
+
+
+def test_get_chunk(mandelbrot_arrays, max_level):
+    scale = max_level - 1
+    virtual_data = _progressive_loading.VirtualData(
+        mandelbrot_arrays[scale], scale=scale
+    )
+    chunk_slice = (slice(1024, 1536, None), slice(512, 1024, None))
+
+    chunk_widths = (
+        chunk_slice[0].stop - chunk_slice[0].start,
+        chunk_slice[1].stop - chunk_slice[1].start,
+    )
+    real_array = get_chunk(chunk_slice, array=virtual_data)
+
+    assert chunk_widths == real_array.shape
+
+
+def test_visual_depth(make_napari_viewer):
+    """TODO: this test passes if you run it by itself"""
```

Comment:
This test doesn't pass when its run as part of the test suite. It fails with a qt error. But if you run this test alone (i.e. `-k test_visual_depth`) it will run without error. Not sure what's going on with qt